### ISSUE TITLE
`docker node task --filter name=` filters on `service`

### ISSUE THREAD

**Issue by albers** [docker/docker#24108]
Taken from the CLI reference page of `docker node tasks`:

> The `name` filter matches on all or part of a task's name.

This is wrong. The filtering is applied on the **exact service name**, see example below.

I think this is actually more useful than filtering on task names and therefore propose to **change the filter name from `name` to `service`**.

Furthermore, filtering should be extended to also cover substring matches, not only exact matches.

```bash
$ docker service create --name top busybox top
7ieif28g1smw7ppav7wiausmz

# a task with name top.1 was created on this node
$ docker node tasks self
ID                         NAME   SERVICE  IMAGE    LAST STATE          DESIRED STATE  NODE
brscnd76m5eh8l3cvmpt5lkj4  top.1  top      busybox  Running 51 seconds  Running        bd4242800842

# filtering by part of the task's name does not find the task
$ docker node tasks --filter name=to self
ID  NAME  SERVICE  IMAGE  LAST STATE  DESIRED STATE  NODE

# filtering by the task's exact name does not find the task
$ docker node tasks --filter name=top.1 self
ID  NAME  SERVICE  IMAGE  LAST STATE  DESIRED STATE  NODE

# ==> only filtering by the SERVICE's name finds the task <==
$ docker node tasks --filter name=top self
ID                         NAME   SERVICE  IMAGE    LAST STATE              DESIRED STATE  NODE
brscnd76m5eh8l3cvmpt5lkj4  top.1  top      busybox  Running About a minute  Running        bd4242800842

$ docker version
Client:
 Version:      1.12.0-dev
 API version:  1.25
 Go version:   go1.6.2
 Git commit:   83e6197-unsupported
 Built:        Tue Jun 28 20:00:19 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-dev
 API version:  1.25
 Go version:   go1.6.2
 Git commit:   83e6197-unsupported
 Built:        Tue Jun 28 20:00:19 2016
 OS/Arch:      linux/amd64
```

**Comment by albers** [comment]
Related issue: #24112

**Comment by vdemeester** [comment]
@albers as far as I know, `node` and `service` *tasks* share the same filters (same backend code), and thus, `service` is available for the `node tasks` command but it would take the long form of service id.

**Comment by albers** [comment]
> it would take the long form of service id

Ah. Interesting. So there is an undocumented filter `service`. It's very hard to create bash completion for undocumented features. 
Thanks @vdemeester for pointing me at it. I'll add it to completion and docs.
By the way, the `service` filter seems not to work. I'll create an issue for that.

**Comment by albers** [comment]
Just found another undocumented filter: `docker service tasks --filter node`, see #24165.

**Comment by tiborvass** [comment]
Related issue #24152

**Comment by aluzzardi** [comment]
@albers @vdemeester Is this simply a documentation issue?

**Comment by yongtang** [comment]
@aluzzardi @vdemeester This issue will be fixed once #24850 is merged in. Currently #24850 is under review. Please take a look and let me know if there are any issues in #24850.

**Comment by albers** [comment]
@aluzzardi 
> Is this simply a documentation issue?

This depends on how the feature should work.
I'm not sure, but I think the docs are right here, so I'm awaiting @yongtang's fix.

**Comment by aluzzardi** [comment]
@albers you're right - not a docs issue actually, we need to merge the PRs above.

@nishanttotla Can you make sure the PRs get in?

**Comment by aluzzardi** [comment]
@nishanttotla @yongtang @vdemeester Were all the required PRs merged in? Will this be fixed in 1.12.2 or are we targeting a future release?

**Comment by yongtang** [comment]
@aluzzardi PR #24850 is done with code review and is now under docs review. I would expect it to be merged very soon. Once #24850 is merged I think this issue could be closed.

**Comment by vdemeester** [comment]
@aluzzardi its targeted for 1.13 so it's not gonna be fixed for 1.12.2 :sweat:

**Comment by albers** [comment]
Assuming that `docker node task` was replaced by `docker node ps`, it works now:
```bash
$ docker node ps --filter name=to
NAME                             IMAGE    NODE          DESIRED STATE  CURRENT STATE          ERROR
top.1.8j2b4x2fsaaopkvsfiecz03i2  busybox  be0f76265216  Running        Running 4 minutes ago

```
Thanks for fixing!
